### PR TITLE
PHPC-677: Keep pem_file valid for life of mongoc_client_t

### PR DESCRIPTION
**Pull request by jmikola**
https://jira.mongodb.org/browse/PHPC-677

### CODE REVIEW

**Comment by jmikola** [comment]
Currently has a memory leak on master due to `_mongoc_ssl_extract_subject()` not freeing the extracted username. I've reported that upstream and it should be included in the fix with [CDRIVER-1209](https://jira.mongodb.org/browse/CDRIVER-1209).

File: php_phongo.c [hunk 1]
```diff
@@ -1948,21 +1948,23 @@ static mongoc_client_t *php_phongo_make_mongo_client(const mongoc_uri_t *uri, zv
...
 		if ((pem = php_stream_context_get_option(ctx, "ssl", "local_cert")) != NULL) {
+			zend_string *s = zval_get_string(pem);
 #else
 		if (SUCCESS == php_stream_context_get_option(ctx, "ssl", "local_cert", &pem)) {
+			convert_to_string_ex(pem);
 #endif
-			char filename[MAXPATHLEN];
+			/* mongoc_client_set_ssl_opts() copies mongoc_ssl_opt_t shallowly;
+			 * its strings must be kept valid for the life of mongoc_client_t */
+			manager->pem_file = ecalloc(1, MAXPATHLEN);
```
Comment (derickr) [inline php_phongo.c]: You seem to allocate it here, but you don't set it to anything?

Comment (jmikola) [inline php_phongo.c]: `manager->pem_file` is set via the `VCWD_REALPATH()` macro below.

**Comment by derickr** [comment]
LGTM

**Pull request merged**